components/cache/adapters/redis_adapter.rst

```diff
@@ -80,6 +80,16 @@ Below are common examples of valid DSNs showing a combination of available value
     // socket "/var/run/redis.sock" and SASL user "user1" and pass "bad-pass"
     RedisAdapter::createConnection('redis://user1:bad-pass@/var/run/redis.sock');
 
+    // a single DSN can define multiple servers using the following syntax:
+    // host[hostname-or-IP:port] (where port is optional). Sockets must include a trailing ':'
+    RedisAdapter::createConnection(
+        'redis:?host[localhost]&host[localhost:6379]&host[/var/run/redis.sock:]=3'
```

Comment:
Wow this is complex, I have a few questions:

- Where does `redis:` come from? Is it the `redis` user? (if so, let's use `user1` as used above)
- Don't we need `redis://` in this syntax?
- How do you configure users or passwords in here, like above?
- Am I correct that `=3` means there were 3 hosts specified in this DNS?

I think we can clarify this a bit more in the comment above or the code example.

Reply:
Hey. I understand that it is complex. I think so too. I've been researching a bit now:


1) `redis:` is our "symfony protocol" for this. We do the same with `memcache:`. `redis://` means one host. `redis:` mean many hosts. 
2) No, we dont. We parse the DNS ourselves with some great regex. 
3) Actually. Redis dont support usernames =). You can configure the `auth` (and many other options) with the second parameters to the `RedisAdapter::createConnection()`. I have added a small example to show how to add `auth` in the DSN.
4) You are, but that is only for `Memcached::createConnection` if I'm correct.

So short answer is. No. I dont think we should clarify the docs more. At least not this example.

Reply:
this is standard url parsing:
`redis:` is the scheme, `//` starts the authority part (ie `auth@example.com`), then is the path, the query and the fragment.
See e.g. https://3v4l.org/ZVCHM or https://3v4l.org/PHau4

Reply:
I thought I knew everything about URLs already. =) Thank you.